src/pages/patientView/ctDNA/HeatmapVAFEvents.tsx

```diff
@@ -96,17 +37,62 @@ export default class HeatmapVAFEvents extends React.Component<IHeatmapVAFEventsP
       let mutSample = mutation.sampleId;
       if (~sampleOrder.indexOf(mutSample) && ~heatmapGeneList.indexOf(mutGene)) {
         let vaf = mutation.tumorAltCount/(mutation.tumorRefCount + mutation.tumorAltCount);
-        let mutPush = {
-          x: sampleOrder.indexOf(mutSample),
-          y: heatmapGeneList.indexOf(mutGene),
-          z: Math.round(vaf * 1000),
-        };
+        let mutPush = [
+          sampleOrder.indexOf(mutSample),
+          heatmapGeneList.indexOf(mutGene),
+          vaf,
+        ];
         plotData.push(mutPush);
       }
     });
 
+    let plotMaker = {
+      data: plotData,
+      xTicks: sampleOrder,
+      yTicks: heatmapGeneList,
+    };
+    return plotMaker;
+  }
+
+  componentDidMount() {
+    let plotMaker = this.extractPlotData();
```

Comment:
If you do this in component did mount, you should make sure that this component cannot re-render in the future, as that could wipe out the canvas.   you should put the call to plot in componentDidUpdate.   that way it could re-execute if necessary.  then implement shouldComponentUpdate to determine if re-render/plot is actually necessary.